ArmPkg/Library/ArmMmuLib/AArch64/ArmMmuLibCore.c

```diff
@@ -274,7 +306,10 @@ UpdateRegionMappingRecursive (
         // No table entry exists yet, so we need to allocate a page table
         // for the next level.
         //
-        TranslationTable = AllocatePages (1);
+        // MU_CHANGE START: Use reserved page table memory pool
+        // TranslationTable = AllocatePages (1);
+        TranslationTable = AllocatePageTableMemory (1);

```

Comment:
One question here: I know I advocated for a different approach in your other PR, but with the current approach of having memory attributes in ApplyMemoryProtectionPolicy, could we not just call AllocatePoolPages here, which skips the memory attribute application? I think preallocating is a good idea, but if we don't change the implementation of free memory RP, do we need it?

Reply:
I think CoreAllocatePoolPages is internal to DxeCore. Is it exposed somehow?

Reply:
Ah, good point. This is non-core code, because is core about manipulating the page table :)

Reply:
This library is both convenient and dangerous 😆